pkg/dashboard/assets/js/self-identification.js

```diff
@@ -0,0 +1,68 @@
+(function () {
+  const emailBoxId = "email-box";
+  const emailInputId = "email-box__email-input";
+  const emailCheckboxId = "emailbox__checkbox";
+  const submitBtnId = "email-box__submit-btn";
+
+  const emailBox = document.getElementById(emailBoxId);
+  const emailInput = document.getElementById(emailInputId);
+  const emailCheckbox = document.getElementById(emailCheckboxId);
+  const submitBtn = document.getElementById(submitBtnId);
+
+  const urlParams = new URLSearchParams(window.location.search);
+
+  initQueryParams();
+  initUIState();
+
+  function initQueryParams() {
+    const enteredEmail = localStorage.getItem("enteredEmail");
+    if (enteredEmail && !urlParams.has("emailEntered")) {
+      setQueryParam("emailEntered", "true");
+    }
+  }
+
+  function setQueryParam(key, value) {
+    urlParams.set(key, value);
+    window.location.search = urlParams;
+  }
+
+  function initUIState() {
+    if (urlParams.get("emailEntered")) {
+      emailBox.style.display = "none";
+    }
+  }
+
+  emailInput.addEventListener("input", function (evt) {
+    const checked = emailCheckbox.checked;
+    toggleSubmitBtn(this.value, checked);
+  });
+
+  emailCheckbox.addEventListener("change", function () {
+    const email = emailInput.value;
+    toggleSubmitBtn(email, this.checked);
+  });
+
+  function toggleSubmitBtn(email, checked) {
+    if (isInputInfoValid(email, checked)) {
+      submitBtn.disabled = false;
+      submitBtn.classList.add("email-box__submit-btn--active");
+    } else {
+      submitBtn.disabled = true;
+      submitBtn.classList.remove("email-box__submit-btn--active");
+    }
+  }
```

Comment:
`<input>` elements can be given the `required` attribute ([MDN Web Docs: `<input>`: The Input (Form Input) element: `required`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#required)), which, when used in a `<form>`, will natively handle this logic. Is there a reason to prefer this JavaScript solution instead?

Reply:
According to the design, we have two states of the submit button. Its default state is `disabled`. When the users are typing or clicking on the checkbox, if the input values are valid, the button will be enabled. Therefore, it will have a different background color and font color. Therefore, we need to use Javascript to check the input values, and then add/remove the `active` class to/from the button. If you have another solution, please feel free to let me know. Thank you

Reply:
If the `required` attribute is added to the `<input>` elements, the submit button will be clickable. However, when clicked, if either of the `<input>` elements is empty, the form won't actually submit. Instead, error messages will pop up instructing the user to fill out the missing fields.

I can see that this behavior doesn't quite match the design specification you have, but I believe that it accomplishes the same end goal and in a simpler way. I think it might be worth considering an update to the design specification here.

Reply:
Thank you for your feedback!
@ddakoda : What do you think?

Reply:
@hieptl I like @thewildmage's feedback here. Agreed we should avoid the JavaScript. 

To accomplish the designs, could we just have a CSS selector for `input[disabled]`?

Reply:
I don't think `input[disabled]` will help since those fields wouldn't be disabled. However, if they're not filled out when they should be, they'll fall under the CSS selector `input:invalid`, like in the demo shown at [MDN Docs: `:invalid`](https://developer.mozilla.org/en-US/docs/Web/CSS/:invalid).

Reply:
I updated the code based on your feedback. The attached video demonstrates the result. Thank y'all!

https://user-images.githubusercontent.com/69354317/184943831-c85b76f6-08c3-4499-bf87-553d0fd6f342.mov

Reply:
Awesome! Thank you!